.github/workflows/firebase_actions.yml

```diff
@@ -25,8 +25,29 @@ jobs:
       - name: Install Firebase CLI
         run: curl -sL https://firebase.tools | bash
 
-      - name: NPM install
-        run: npm install
+      - name: Install NPM dependencies
+        run: npm install && npm install --prefix functions
+        working-directory: metrics/firebase/
+
+      - uses: cedx/setup-dart@v2
+        with:
+          release-channel: beta
+          version: 2.12.0-133.2.beta
+
+      - name: Get dependencies
+        run: pub get
+        working-directory: metrics/firebase/functions/
+
+      - name: Dart analyze
+        run: dartanalyzer . --fatal-infos --fatal-warnings
+        working-directory: metrics/firebase/functions/
+
+      - name: Run Cloud Functions tests
+        run: pub run test
+        working-directory: metrics/firebase/functions/
+
+      - name: Deploy Cloud Functions
+        run: firebase use default && firebase deploy --only functions
```

Comment:
Would it be great to deploy functions and rules at once using the `firebase deploy --only functions,firestore:rules` command?